packages/curve-ui-kit/src/shared/ui/Balance.tsx

```diff
@@ -0,0 +1,106 @@
+import AccountBalanceWalletOutlinedIcon from '@mui/icons-material/AccountBalanceWalletOutlined'
+import Button from '@mui/material/Button'
+import Stack from '@mui/material/Stack'
+import Typography from '@mui/material/Typography'
+import { SizesAndSpaces } from '@ui-kit/themes/design/1_sizes_spaces'
+
+const { Spacing, IconSize } = SizesAndSpaces
+
+const formatNumber = (value: number): string =>
+  value.toLocaleString(undefined, {
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  })
+
+type MaxButtonProps = {
+  children: React.ReactNode
+  underline: boolean
+  onClick?: () => void
+}
+
+/** Reusable Max button component with consistent styling */
+const MaxButton = ({ children, underline, onClick }: MaxButtonProps) => (
+  <Button
+    color="ghost"
+    variant="link"
+    size="extraSmall"
+    onClick={onClick}
+    sx={{
+      minWidth: 'unset',
+      // Sadly the !important is still necessary because of the !important in defineMuiButton for the Link variant :(
```

Comment:
Could we just remove the height from the link variant when size="extraSmall"?

Reply:
Sadly not, because the `ghost` variant [overrides](https://github.com/curvefi/curve-frontend/blob/main/packages/curve-ui-kit/src/themes/components/button/mui-button.ts#L77) the height set by the button size. At the same time we still need to have the `extraSmall` size because of other props like font size, line height, text transform etc.

Reply:
That inconsistency in the design would be a bug IMO - but I checked Figma and the variant should be `buttonXs` not "ghost"

That is a typography variant, not a button color though. We could either introduce that as a button color, wrap the children with `Typography` or add onClick to the `Typography` instead of `Button`

Reply:
Btw I think the [override](https://github.com/curvefi/curve-frontend/blob/main/packages/curve-ui-kit/src/themes/components/button/mui-button.ts#L77) is just plain wrong - why should the ghost override the size

Reply:
https://github.com/curvefi/curve-frontend/pull/962/files/f82dbbf7da69d549bb72d827e9b3445567aba4d1#r2097241632